src/main/java/org/commcare/formplayer/objects/SerializableMenuSession.java

```diff
@@ -1,30 +1,61 @@
-package org.commcare.formplayer.repo;
+package org.commcare.formplayer.objects;
 
 import org.commcare.formplayer.session.MenuSession;
+import org.hibernate.annotations.GenericGenerator;
+import org.springframework.data.jpa.domain.support.AuditingEntityListener;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
+
+import javax.persistence.*;
 
 /**
  * Created by willpride on 8/1/16.
  */
+
+@Entity
+@Table(name="menu_sessions")
+@EntityListeners(AuditingEntityListener.class)
 public class SerializableMenuSession {
 
+    @Id @GeneratedValue(generator="system-uuid")
+    @GenericGenerator(name="system-uuid", strategy = "uuid")
     private String uuid;
+
+    @Column(name="username")
     private String username;
+
+    @Column(name="domain")
     private String domain;
+
+    @Column(name="appId")
     private String appId;
+
+    @Column(name="installReference")
     private String installReference;
+
+    @Column(name="locale")
     private String locale;
+
+    // Todo: Will this be large, should this be @Lob?
+    @Column(name="commcareSession")
     private byte[] commcareSession;
+
+    @Column(name="asUser")
     private String asUser;
+
+    @Column(name="oneQuestionPerScreen")
```

Comment:
interestingly this isn't a persisted field, looking at it's usage I think it can be removed form this class altogether and kept purely in the `MenuSession` class. The only time this might break is if there is end of form navigation that advances to another form in which case I don't think the `oneQuestionPerScreen` value will get passed on. I don't think that's worth worrying about at this point.

Reply:
Can we also simply use `@Transient` instead of moving it somewhere else?  https://docs.oracle.com/javaee/7/api/javax/persistence/Transient.html

Reply:
yes we could do that though in the long run it would be better to move it out